test/tests/fips_test_suite/self_tests.rs

```diff
@@ -0,0 +1,103 @@
+// Licensed under the Apache-2.0 license
+use crate::common;
+
+use caliptra_builder::firmware::ROM_WITH_UART_FIPS_TEST_HOOKS;
+use caliptra_drivers::CaliptraError;
+use caliptra_drivers::FipsTestHook;
+use caliptra_hw_model::{BootParams, HwModel, InitParams};
+use common::*;
+
+#[test]
+#[cfg(not(feature = "test_env_immutable_rom"))]
+pub fn kat_halt_check_no_output() {
+    let rom = caliptra_builder::build_firmware_rom(&ROM_WITH_UART_FIPS_TEST_HOOKS).unwrap();
+
+    let boot_params = BootParams {
+        init_params: InitParams {
+            rom: &rom,
+            ..Default::default()
+        },
+        initial_dbg_manuf_service_reg: (FipsTestHook::HALT_SELF_TESTS as u32) << HOOK_CODE_OFFSET,
+        ..Default::default()
+    };
+
+    let mut hw = fips_test_init_to_boot_start(Some(boot_params));
+
+    // Wait for ACK that ROM reached halt point
+    hook_wait_for_complete(&mut hw);
+
+    // Check output is inhibited
+    verify_output_inhibited(&mut hw);
+
+    // TODO: Remove continuing if it's not needed
+    // Tell ROM to continue
+    hook_code_write(&mut hw, FipsTestHook::CONTINUE);
+
+    // Wait for ACK that ROM continued
+    hook_wait_for_complete(&mut hw);
+
+    // Step to ready for FW in ROM
+    hw.step_until(|m| m.soc_ifc().cptra_flow_status().read().ready_for_fw());
+}
+
+#[test]
+#[cfg(not(feature = "test_env_immutable_rom"))]
+pub fn kat_sha384_error() {
+    let rom = caliptra_builder::build_firmware_rom(&ROM_WITH_UART_FIPS_TEST_HOOKS).unwrap();
+
+    let boot_params = BootParams {
+        init_params: InitParams {
+            rom: &rom,
+            ..Default::default()
+        },
+        initial_dbg_manuf_service_reg: (FipsTestHook::SHA384_ERROR as u32) << HOOK_CODE_OFFSET,
+        ..Default::default()
+    };
+
+    let mut hw = fips_test_init_to_boot_start(Some(boot_params));
+
+    // Wait for fatal error
+    hw.step_until(|m| m.soc_ifc().cptra_fw_error_fatal().read() != 0);
+
+    // Verify fatal code is correct
+    assert_eq!(
+        hw.soc_ifc().cptra_fw_error_fatal().read(),
+        u32::from(CaliptraError::ROM_KAT_SHA384_DIGEST_MISMATCH)
+    );
+
+    // TODO: Verify we cannot use the algorithm
```

Comment:
All of these test TODOs will be completed in a later PR with these full test implementations and additional tests. These partial tests were just included to demonstrate the usage of the FIPS test hooks to the point where the correct error code is received from the injected error.